packages/blocks/src/store/selectors.js

```diff
@@ -96,3 +132,22 @@ export const getChildBlockNames = createSelector(
 export const hasChildBlocks = ( state, blockName ) => {
 	return getChildBlockNames( state, blockName ).length > 0;
 };
+
+/**
+ * Returns a boolean indicating if a block has at least one child block with inserter support.
+ *
+ * @param {Object} state     Data state.
+ * @param {string} blockName Block type name.
+ *
+ * @return {boolean} True if a block contains at least one child blocks with inserter support and false otherwise.
+ */
+export const hasChildBlocksWithInserterSupport = createSelector(
+	( state, blockName ) => {
+		return some( getChildBlockNames( state, blockName ), ( currentBlockName ) => {
```

Comment:
Minor: Would `child` make more sense as a prefix in place of `current` ? It wasn't immediately clear to me what's "current" about what we're iterating.